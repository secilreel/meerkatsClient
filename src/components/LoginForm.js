import React, {Component} from 'react';
import AuthApiService from '../services/auth-api-services';
import TokenService from '../services/token-service'; 
import {Link} from 'react-router-dom'

export default class LoginForm extends Component{
    static defaultProps = {
        onLoginSuccess: () => {}
    }
    state ={error: null};

    handleSubmitJWTAuth = e =>{
        e.preventDefault();
        this.setState({error: null});
        const {user_name, password} = e.target;

        AuthApiService.postLogin({
            user_name: user_name.value,
            password: password.value,
        })
        .then(res => {
            user_name.value = '';
            password.value = '';
            TokenService.saveAuthToken(res.authToken);
            this.props.onLoginSuccess();
        })
        .catch(res => {
            this.setState({error: res.error});
        })
    }

    render(){
        return(
            <div className="login container">
                <h2>Login</h2>
                <form onSubmit={
                    this.handleSubmitJWTAuth}>
                    <div className="login-box">
                    <div className="username">
                        <label htmlFor="loginUsername">User name:</label>
                        <input required name="user_name"/>
                    </div>
                    <div className="password">
                        <label htmlFor="loginPassword">Password:</label>
                        <input required name="password" type="password"/>
                    </div>
                    <div className="button-box">
                    <button type="submit" value="Submit">Login</button>
                    </div>
                    <div className="register">
                    <Link to={'/register'}>
                    <button type="submit" value="Submit">Register Here!</button>
                    </Link>
                    </div>
                    </div>
                </form>
            </div>
        );
    }
}