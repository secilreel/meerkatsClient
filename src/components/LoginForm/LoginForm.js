import React, {Component} from 'react';
import AuthApiService from '../../services/auth-api-services';
import TokenService from '../../services/token-service'; 

export default class LoginForm extends Component{
    static defaultProps = {
        onLoginSuccess: () => {}
    }
    state ={error: null};

    handleSubmitJWTAuth = e =>{
        e.preventDefault();
        this.setState({error: null});
        const {user_name, password} = e.target;
        console.log(user_name.value);
        console.log(password.value);

        AuthApiService.postLogin({
            user_name: user_name.value,
            password: password.value,
        })
        .then(res => {
            user_name.value = '';
            password.value = '';
            TokenService.saveAuthToken(res.authToken);
            this.props.onLoginSuccess();
        })
        .catch(res => {
            this.setState({error: res.error});
        })

    }

    render(){
        return(
            <div className="login container">
                <h2>Login</h2>
                <form onSubmit={
                    this.handleSubmitJWTAuth}>
                    <div className="login-box">
                    <div className="username">
                        <label htmlFor="loginUsername">User name:</label>
                        <input required name="user_name"/>
                    </div>
                    <div className="password">
                        <label htmlFor="loginPassword">Password:</label>
                        <input required name="password" type="password"/>
                    </div>
                    <div className="button-box">
                    <button type="submit" value="Submit">Login</button>
                    </div>
                    </div>
                </form>
            </div>
        );
    }
}