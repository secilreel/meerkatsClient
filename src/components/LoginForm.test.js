import React from 'react';
import ReactDOM from 'react-dom';
import LoginForm from './LoginForm';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

configure({ adapter: new Adapter() });

describe('<LoginForm />', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <LoginForm />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
    <MemoryRouter>
    <LoginForm />
    </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });

it('renders empty given no selection', () => {
    const wrapper = shallow(<LoginForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

it('Should call props.onClick() when the Close button is clicked', () => {
    const fakeEvent = { preventDefault: () => console.log('preventDefault') };
    const clickHandler = () => {console.log('button clicked')};
    const wrapper = shallow(<LoginForm onClick={clickHandler}/>);
    wrapper.find('button').at(0).simulate('click', fakeEvent);
    expect(toJson(wrapper)).toMatchSnapshot();
});
})
