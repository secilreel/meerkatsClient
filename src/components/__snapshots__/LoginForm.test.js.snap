// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> Should call props.onClick() when the Close button is clicked 1`] = `
<div
  className="login container"
>
  <h2>
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="login-box"
    >
      <div
        className="username"
      >
        <label
          htmlFor="loginUsername"
        >
          User name:
        </label>
        <input
          name="user_name"
          required={true}
        />
      </div>
      <div
        className="password"
      >
        <label
          htmlFor="loginPassword"
        >
          Password:
        </label>
        <input
          name="password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="button-box"
      >
        <button
          type="submit"
          value="Submit"
        >
          Login
        </button>
      </div>
      <div
        className="register"
      >
        <Link
          replace={false}
          to="/register"
        >
          <button
            type="submit"
            value="Submit"
          >
            Register Here!
          </button>
        </Link>
      </div>
    </div>
  </form>
</div>
`;

exports[`<LoginForm /> renders empty given no selection 1`] = `
<div
  className="login container"
>
  <h2>
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="login-box"
    >
      <div
        className="username"
      >
        <label
          htmlFor="loginUsername"
        >
          User name:
        </label>
        <input
          name="user_name"
          required={true}
        />
      </div>
      <div
        className="password"
      >
        <label
          htmlFor="loginPassword"
        >
          Password:
        </label>
        <input
          name="password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="button-box"
      >
        <button
          type="submit"
          value="Submit"
        >
          Login
        </button>
      </div>
      <div
        className="register"
      >
        <Link
          replace={false}
          to="/register"
        >
          <button
            type="submit"
            value="Submit"
          >
            Register Here!
          </button>
        </Link>
      </div>
    </div>
  </form>
</div>
`;

exports[`<LoginForm /> renders the UI as expected 1`] = `
<div
  className="login container"
>
  <h2>
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <div
      className="login-box"
    >
      <div
        className="username"
      >
        <label
          htmlFor="loginUsername"
        >
          User name:
        </label>
        <input
          name="user_name"
          required={true}
        />
      </div>
      <div
        className="password"
      >
        <label
          htmlFor="loginPassword"
        >
          Password:
        </label>
        <input
          name="password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="button-box"
      >
        <button
          type="submit"
          value="Submit"
        >
          Login
        </button>
      </div>
      <div
        className="register"
      >
        <a
          href="/register"
          onClick={[Function]}
        >
          <button
            type="submit"
            value="Submit"
          >
            Register Here!
          </button>
        </a>
      </div>
    </div>
  </form>
</div>
`;
